qutip_lattice/lattice.py

```diff
@@ -200,1108 +257,6196 @@ def cell_structures(val_s=None, val_t=None, val_u=None):
+            ax.plot([x_cell-1+self.positions_of_sites[csn-1],
+                     x_cell+self.positions_of_sites[0]], [0.0, 0.0],
+                    "-", c="r", lw=1.5, zorder=7)
 
-    def k(self):
-        """
-        Returns the crystal momentum operator. All degrees of freedom has the
-        cell number at their correspondig entry in the position operator.
+            x_cell = self.num_cell
+            x_b = 2*x_cell-1+self.positions_of_sites[csn-1]
+            x_b = (x_b+self.positions_of_sites[0])/2
 
-        Returns
-        -------
-        Qobj(ks) : qutip.Qobj
-            The crystal momentum operator in units of 1/a. L is the number
-            of unit cells, a is the length of a unit cell which is always taken
-            to be 1.
-        """
-        L = self.num_cell
-        kop = np.zeros((L, L), dtype=complex)
-        for row in range(L):
-            for col in range(L):
-                if row == col:
-                    kop[row, col] = (L-1)/2
-#                    kop[row, col] = ((L+1) % 2)/ 2
-                    # shifting the eigenvalues
-                else:
-                    kop[row, col] = 1/(np.exp(2j * np.pi * (row - col)/L) - 1)
-        qkop = Qobj(kop, type='oper')
-        [kD, kV] = qkop.eigenstates()
-        kop_P = np.zeros((L, L), dtype=complex)
-        for eno in range(L):
-            if kD[eno] > (L // 2 + 0.5):
-                vl = kD[eno] - L
-            else:
-                vl = kD[eno]
-            vk = kV[eno]
-            kop_P = kop_P + (vl * vk.proj()).full()
-        kop = 2 * np.pi / L * kop_P
-        nx = self.cell_num_site
-        ne = self._length_for_site
-        k = np.kron(kop, np.eye(nx*ne))
-        dim_H = [self.lattice_tensor_config, self.lattice_tensor_config]
-        return Qobj(k, dims=dim_H, type='oper')
+            plt.text(x=x_b, y=0.1, s='T', horizontalalignment='center',
+                     verticalalignment='center')
+            ax.plot([x_cell-1+self.positions_of_sites[csn-1],
+                     x_cell+self.positions_of_sites[0]], [0.0, 0.0],
+                    "-", c="r", lw=1.5, zorder=7)
+
+        x2 = (1+self.positions_of_sites[csn-1])/2
+        x1 = x2-1
+        h = 0.5
+
+        if self.num_cell > 2:
+            xu = 1    # The index of cell over which the black box is drawn
+            x1 = x1+xu
+            x2 = x2+xu
+        ax.plot([x1, x1], [-h, h], "-", c="k", lw=1.5, zorder=7, alpha=0.3)
+        ax.plot([x2, x2], [-h, h], "-", c="k", lw=1.5, zorder=7, alpha=0.3)
+        ax.plot([x1, x2], [h, h], "-", c="k", lw=1.5, zorder=7, alpha=0.3)
+        ax.plot([x1, x2], [-h, -h], "-", c="k", lw=1.5, zorder=7, alpha=0.3)
+        plt.axis('off')
+        plt.show()
+        plt.close()
+        dim_site = list(np.delete(self.cell_tensor_config, [0], None))
+        dims_site = [dim_site, dim_site]
+        return Qobj(inter_T, dims=dims_site, type='oper')
+
+class Lattice1d_fermi_Hubbard():
```

Comment:
This should inherit from `Lattice1d`.

This should simplify both the code and users experience.